directives/wfm-multiple-search/wfm-multiple-search-input.directive.js

```diff
@@ -0,0 +1,140 @@
+
+'use strict';
+
+	var multipleSearchInputCtrl = function () {
+    var vm = this;
+    vm.advancedSearchForm = {};
+
+    var searchExpressionSeprator = ';';
+    var keyValueSeprator = ':';
+
+    vm.validateSearchKeywordChanged = function () {
+        vm.searchOptions.searchKeywordChanged = true;
+        parseSearchExpressionInputted();
+    };
+    vm.turnOffAdvancedSearch = function () {
+        vm.showAdvancedSearchOption = false;
+    };
+
+    vm.toggleAdvancedSearchOption = function ($event) {
+        vm.showAdvancedSearchOption = !vm.showAdvancedSearchOption;
+        $event.stopPropagation();
+        parseSearchExpressionInputted();
+    };
+    vm.advancedSearch = function () {
+        vm.showAdvancedSearchOption = false;
+
+        var expression = '';
+        angular.forEach(vm.searchOptions.searchFields, function (searchType) {
+            var title = searchType.charAt(0).toLowerCase() + searchType.slice(1);
+            expression += getSearchCriteria(title, vm.advancedSearchForm[searchType]);
+        });
+
+        expression = expression.trim();
+        if (expression !== '' && expression !== vm.searchOptions.keyword) {
+            vm.searchOptions.searchKeywordChanged = true;
+        }
+        vm.searchOptions.keyword = expression;
+        if (vm.searchCallback) {
+            vm.searchCallback(expression);
+        }
+    };
+
+    function setSearchFormProperty(searchType, searchValue) {
+        angular.forEach(vm.searchOptions.searchFields, function (propName) {
+            if (propName.toUpperCase() === searchType.toUpperCase()) {
+                vm.advancedSearchForm[propName] = searchValue;
+            }
+        });
+    }
+
+    function parseSearchValue(value) {
+        if (value === undefined || value === null || value.trim().length === 0) {
+            return '';
+        }
+        var displayValue = value.trim();
+
+        var quotedKeywords = '';
+        var pattern = /['"](.*?)['"]/ig;
+        var match;
+        while (match === pattern.exec(displayValue)) {
+            var keyword = match[1].trim();
+            if (keyword.length > 0) {
+                quotedKeywords = quotedKeywords + ' "' + keyword + '"';
+            }
+        }
+        quotedKeywords = quotedKeywords.trim();
+
+        var unquotedKeywords = displayValue
+				.replace(pattern, '').trim()
+				.replace('"', '').trim();
+
+        return (quotedKeywords + ' ' + unquotedKeywords).trim();
+    }
+
+    function parseSearchExpressionInputted() {
+        vm.advancedSearchForm = {};
+        if (vm.searchOptions.keyword.indexOf(keyValueSeprator) !== -1) {
+            var expression = vm.searchOptions.keyword.trim();
+            if (expression.charAt(expression.length - 1) !== searchExpressionSeprator) {
+                expression = expression + searchExpressionSeprator;
+            }
+            var regex = /(\S*?):\s{0,}(.*?);/ig;
+            var match;
+            while (match === regex.exec(expression)) {
```

Comment:
If you just wrap it with parentheses as @shenba2014 suggests, the lint will see it as an expression.